src/Psalm/Internal/Provider/ReturnTypeProvider/ArrayMapReturnTypeProvider.php

```diff
@@ -505,7 +504,7 @@ public static function getReturnTypeFromMappingIds(
     public static function cleanContext(Context $context, int $fake_var_discriminator): void
     {
         foreach ($context->vars_in_scope as $var_in_scope => $_) {
-            if (str_contains($var_in_scope, "__fake_{$fake_var_discriminator}_")) {
```

Comment:
Psalm did not pick that up? Polyfills are messing with the analysis maybe?

I wonder how hard it could be to add a new config forbidPolyfills that, when enabled would warn about functions removed through deltas but defined elsewhere...

Reply:
> Polyfills are messing with the analysis maybe?

Reflection, likely.